screen-write.c

```diff
@@ -653,34 +662,45 @@ screen_write_menu(struct screen_write_ctx *ctx, struct menu *menu,
 	cx = s->cx;
 	cy = s->cy;
 
-	memcpy(&default_gc, &grid_default_cell, sizeof default_gc);
+	memcpy(&default_gc, menu_gc, sizeof default_gc);
 
 	screen_write_box(ctx, menu->width + 4, menu->count + 2,
-	    BOX_LINES_DEFAULT, &default_gc, menu->title);
+	    BOX_LINES_DEFAULT, border_gc, menu->title);
 
 	for (i = 0; i < menu->count; i++) {
 		name = menu->items[i].name;
 		if (name == NULL) {
+			/* Draw separator line */
 			screen_write_cursormove(ctx, cx, cy + 1 + i, 0);
-			screen_write_hline(ctx, menu->width + 4, 1, 1);
-		} else {
-			if (choice >= 0 && i == (u_int)choice && *name != '-')
-				gc = choice_gc;
-			screen_write_cursormove(ctx, cx + 2, cy + 1 + i, 0);
-			for (j = 0; j < menu->width; j++)
-				screen_write_putc(ctx, gc, ' ');
-			screen_write_cursormove(ctx, cx + 2, cy + 1 + i, 0);
-			if (*name == '-') {
-				name++;
-				default_gc.attr |= GRID_ATTR_DIM;
-				format_draw(ctx, gc, menu->width, name, NULL,
-				    0);
-				default_gc.attr &= ~GRID_ATTR_DIM;
-			} else
-				format_draw(ctx, gc, menu->width, name, NULL,
-				    gc == choice_gc);
-			gc = &default_gc;
+			screen_write_hline(ctx, menu->width + 4, 1, 1,
+			    // TODO: Are separator lines considered to be part
+			    // of the menu or rather the menu border? In the
+			    // latter case they need be drawn with border_gc
+			    // rather than &default_gc?
+			    &default_gc, border_gc);
```

Comment:
Are separator lines considered to be part of the menu? 

| Yes: `&default_gc, border_gc);` | No: `border_gc, border_gc);`
| --- | ---
| ![image](https://github.com/tmux/tmux/assets/16507/8942d14d-cfbe-418d-9f75-9df1996c4c89) | ![image](https://github.com/tmux/tmux/assets/16507/d57507d0-1395-47b8-9416-d21abbb873fd)